lib/spack/llnl/util/filesystem.py

```diff
@@ -568,20 +568,22 @@ def find_libraries(args, root, shared=True, recurse=False):
     """Returns an iterable object containing a list of full paths to
     libraries if found.
 
-    Args:
-        args: iterable object containing a list of library names to \
-            search for (e.g. 'libhdf5')
-        root: root folder where to start searching
-        shared: if True searches for shared libraries, otherwise for static
-        recurse: if False search only root folder, if True descends top-down \
-            from the root
-
-    Returns:
-        list of full paths to the libraries that have been found
+    :param args: Library name(s) to search for
+    :type args: str or collections.Sequence
+    :param str root: The root directory to start searching from
+    :param bool shared: if True searches for shared libraries,
+        otherwise for static
+    :param bool recurse: if False search only root folder,
+        if True descends top-down from the root
+
+    :returns: The libraries that have been found
+    :rtype: LibraryList
     """
-    if not isinstance(args, collections.Sequence) or isinstance(args, str):
-        message = '{0} expects a sequence of strings as first argument'
-        message += ' [got {1} instead]'
+    if isinstance(args, str):
+        args = [args]
```

Comment:
also, I would prefer if we split this PR into two: 

1. Change the function and add unit tests to make sure it works as expected
2. Apply it to the packages.